src/Executor.php

```diff
@@ -9,7 +9,7 @@ class Executor
     /**
      * @var Purger
      */
-    private $purger;
+    private Purger $purger;
```

Comment:
is this php 7 compatible? If not, we probably should bump the php version too

Reply:
Nice spotting! I just checked it, looks like it is 7.4 compatible according to [this](https://wiki.php.net/rfc/typed_properties_v2), so we should bump the php version from ^7.3 to ^7.4?